feature-libs/cart/base/components/clear-cart/clear-cart-dialog/clear-cart-dialog-component.service.spec.ts

```diff
@@ -0,0 +1,96 @@
+import { TestBed } from '@angular/core/testing';
+import { ActiveCartFacade, OrderEntry } from '@spartacus/cart/base/root';
+import { Observable, of } from 'rxjs';
+import { ClearCartDialogComponentService } from './clear-cart-dialog-component.service';
+import { LaunchDialogService } from '@spartacus/storefront';
+import { GlobalMessageService, GlobalMessageType } from '@spartacus/core';
+import { tap } from 'rxjs/operators';
+import createSpy = jasmine.createSpy;
+
+const mockCloseReason = 'Close Dialog';
+const mockCartEntry: OrderEntry = { entryNumber: 0 };
+
+class MockGlobalMessageService implements Partial<GlobalMessageService> {
+  add = createSpy().and.stub();
+}
+
+class MockLaunchDialogService implements Partial<LaunchDialogService> {
+  closeDialog(_reason: string): void {}
+}
+
+class MockActiveCartFacade implements Partial<ActiveCartFacade> {
+  getEntries(): Observable<OrderEntry[]> {
+    return of();
+  }
+}
+
+describe('ClearCartDialogComponentService', () => {
+  let service: ClearCartDialogComponentService;
+  let activeCartFacade: ActiveCartFacade;
+  let launchDialogService: LaunchDialogService;
+  let globalMessageService: GlobalMessageService;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      providers: [
+        { provide: LaunchDialogService, useClass: MockLaunchDialogService },
+        { provide: ActiveCartFacade, useClass: MockActiveCartFacade },
+        {
+          provide: GlobalMessageService,
+          useClass: MockGlobalMessageService,
+        },
+      ],
+    }).compileComponents();
+
+    service = TestBed.inject(ClearCartDialogComponentService);
+    launchDialogService = TestBed.inject(LaunchDialogService);
+    activeCartFacade = TestBed.inject(ActiveCartFacade);
+    globalMessageService = TestBed.inject(GlobalMessageService);
+  });
+
+  it('should be created', () => {
+    expect(service).toBeTruthy();
+  });
+
+  it('should get and change clearing cart progess', () => {
+    expect(service.isClearing$.value).toBeFalsy();
+    service.clearActiveCart();
```

Comment:
This test might need some work.  
After `service.clearActiveCart()` is called, I think the processing is async and if so, `getClearingCartProgess()` could be either true or false.  

If the test's goal is to validate the value of `getClearingCartProgess()`, the test needs to make sure the result is deterministic.

Reply:
Well I think it is deterministic, because the call `service.clearActiveCart()` emits the next value of the observable returned by `getClearingCartProgess()`. And the expect occurs once the observables emits a value